projects/karak/index.js

```diff
@@ -1,22 +1,38 @@
+const { getLogs2 } = require('../helper/cache/getLogs')
 const { sumTokens2 } = require("../helper/unwrapLPs")
 
 const config = {
-  ethereum: { factory: '0x54e44dbb92dba848ace27f44c0cb4268981ef1cc', },
-  arbitrum: { factory: '0x399f22ae52a18382a67542b3de9bed52b7b9a4ad', },
-  karak: { factory: '0xB308474350D75447cA8731B7Ce87c9ee9DA03B1C', },
-  mantle: { factory: '0x4a2b015CcB8658998692Db9eD4522B8e846962eD', },
-  bsc: { factory: '0x4a2b015CcB8658998692Db9eD4522B8e846962eD', },
-  blast: { factory: '0x58b5dc145ca2BE84fe087614CFe36055be609BB3', },
-  fraxtal: { factory: '0xdF922c74CC0dc394022ea002Af5aFaa32348670e', },
+  ethereum: { v1: { factory: '0x54e44dbb92dba848ace27f44c0cb4268981ef1cc' } },
+  arbitrum: { v1: { factory: '0x399f22ae52a18382a67542b3de9bed52b7b9a4ad' }, v2: { factory: '0xc4B3D494c166eBbFF9C716Da4cec39B579795A0d', block: 261874079 }},
+  karak: { v1: { factory:'0xB308474350D75447cA8731B7Ce87c9ee9DA03B1C' } },
+  mantle: { v1: { factory: '0x4a2b015CcB8658998692Db9eD4522B8e846962eD'} },
+  bsc: { v1: { factory: '0x4a2b015CcB8658998692Db9eD4522B8e846962eD'} },
+  blast: { v1: { factory: '0x58b5dc145ca2BE84fe087614CFe36055be609BB3'} },
+  fraxtal: { v1: { factory: '0xdF922c74CC0dc394022ea002Af5aFaa32348670e'}, v2: { factory: '0x04962047B6a9E8c99C8Da874D34c4285a87d541E', block: 11669871 } },
+}
+
+const eventAbi = 'event DeployedVault(address operator, address vault, address asset)'
+
+const karak_v1_tvl = async (api, { factory }) => {
+  const vaults = await api.call({ abi: 'address[]:getVaults', target: factory })
+  const tokens = await api.multiCall({ abi: 'address:asset', calls: vaults })
+  return sumTokens2({ api, tokensAndOwners2: [tokens, vaults] })
+}
+
+const karak_v2_tvl = async (api, { factory, block }) => {
+  const logs = await getLogs2({ api, target: factory, topic: '0x754c0ba87ea068d6b27ef93ca2981913e3a5d0d09ac8fa485b7691e0dd55f180', fromBlock: block, eventAbi })
```

Comment:
topic is not needed in most cases (there are some edge cases/bugs where we dont compute correct topic from eventAbi)